backend/api/messaging/index.js

```diff
@@ -42,9 +43,13 @@ router.post(
     }
 
     const { sid } = clients.getReceiverByChannel(channel, sender);
-    socketEmit('chat-message', sid, dataToPublish);
-
-    return res.send({ message: 'message sent' });
+    const received = await socketEmit('chat-message', sid, dataToPublish);
+    if (received) {
+      const senderID = clients.getSenderByChannel(channel, sender);
+      socketEmit('delivered', senderID.sid, id);
+      return res.send({ message: 'message sent' });
+    }
+    return res.send(400);
```

Comment:
This goes against the idea of using websocket. No point in blocking http response.
As I mentioned earlier, it needs to be handled at the **client-side of the receiving end**.
```suggestion
socketEmit('chat-message', sid, dataToPublish);
return res.send({ message: 'message sent' });
```